src/test/java/io/appium/java_client/appium/element/generation/BaseElementGenerationTest.java

```diff
@@ -34,6 +35,7 @@ protected boolean check(Supplier<DesiredCapabilities> serverCapabilitiesSupplier
                          BiPredicate<By, Class<? extends WebElement>> filter,
                          By by, Class<? extends WebElement> clazz) {
         AppiumServiceBuilder builder = new AppiumServiceBuilder()
+                .withAppiumJS(new File("/usr/local/lib/node_modules/appium/build/lib/main.js"))
```

Comment:
shouldn't this be cross-platform?

Reply:
Sorry didn't get you?

Reply:
we could find npm root by executing `npm root -g`. The second part of the path is always constant and can be added using path joining

Reply:
Yup client does the same internally [here](https://github.com/appium/java-client/blob/master/src/main/resources/scripts/get_path_to_default_node.sh) but not sure why it couldn't find it inside the azure mac host. So i harcoded the same.

Reply:
I would say it would be better to put  appium.js path calculation to a separate helper method rather than hardcoding it

Reply:
yup will do that

Reply:
Something like `Paths.get(<Path_to_npm_root>, "appium", "build" ,"lib", "main.js")`

Reply:
I'm thinking why does client doesn't detect globally installed server path automatically when there is nvm in place in a system... I do remember there are some issues related to it.